mars/dataframe/core.py

```diff
@@ -1837,6 +1837,63 @@ def itertuples(self, index=True, name='Pandas', batch_size=1000, session=None):
         """
         return self._data.itertuples(batch_size=batch_size, session=session,
                                      index=index, name=name)
+    
+    def assign(self, **kwargs):
+        r"""
+        Assign new columns to a DataFrame.
+        Returns a new object with all original columns in addition to new ones.
+        Existing columns that are re-assigned will be overwritten.
+        Parameters
+        ----------
+        **kwargs : dict of {str: callable or Series}
+            The column names are keywords. If the values are
+            callable, they are computed on the DataFrame and
+            assigned to the new columns. The callable must not
+            change input DataFrame (though pandas doesn't check it).
+            If the values are not callable, (e.g. a Series, scalar, or array),
+            they are simply assigned.
+        Returns
+        -------
+        DataFrame
+            A new DataFrame with the new columns in addition to
+            all the existing columns.
+        Notes
+        -----
+        Assigning multiple columns within the same ``assign`` is possible.
+        Later items in '\*\*kwargs' may refer to newly created or modified
+        columns in 'df'; items are computed and assigned into 'df' in order.
+        Examples
+        --------
+        >>> df = pd.DataFrame({'temp_c': [17.0, 25.0]},
```

Comment:
Could you please rewrite the pandas examples into Mars DataFrame, refer to 

https://github.com/mars-project/mars/blob/0c9114d1fffe56af7e3a3da01c7534e9597237a2/mars/dataframe/core.py#L1226-L1237 .

Don't forget to call execute like `df.assign(xxx).execute()` to trigger execution and get result.

Reply:
Sure! I am working on it